src/controllers/book.ts

```diff
@@ -0,0 +1,145 @@
+// Dependencies
+import { Request, Response } from 'express';
+import { IBookQuery } from '../interfaces/book-query';
+
+// Models
+import { Book } from '../models/book';
+
+// Services
+import { verifyUserService } from '../services/verify-user';
+import { getBookService } from '../services/get-book';
+
+// Constant/Enum
+import {
+  BOOK_NON_STRING_QUERIES,
+  BOOK_PATCHABLE_FIELDS,
+  COMMON_QUERIES
+} from '../utils/common/book-filters';
+
+// Utils
+import { toLowerCaseQuery } from '../utils/functions/lower-case-query';
+import { CustomError } from '../utils/classes/custom-error-class';
+import { SortOrder } from 'mongoose';
+
+class BookController {
+  public async createNewBook(req: Request, res: Response): Promise<void> {
+    try {
+      await verifyUserService.isGuest(req, res);
```

Comment:
Hàm này được đặt tên với `is` nên nó phải trả kết quả là boolean và nơi gọi phải sử dụng nó để kiểm tra điều kiện. Còn nếu giả định throw lỗi thì nên đặt tên là `verifyXXX`

Reply:
nên làm rõ

```js
if (await verifyUserService.isGuest(req, res)) {
   res.status(401).....
   return;
}
```